### PR TITLE
Update containers for EL8

### PR DESCRIPTION
**Pull request by anthonygego**
Linked to UCL-INGI/INGInious#770

This updates the container files to make them build with EL8 base container. Some notable changes:
- The gccX containers are updated with available toolkits in EL8.
- All the containers with a compilation process have been cleaned so that, wherever possible, RPM could be used to help solve runtime dependencies and remove the build dependencies.
- Software have been most of the time updated to a more recent version.
- The following containers used at UCLouvain were added:
  * dafny
  * openssh-clients
  * python3-turtle (separated from python3 as it requires the X librairies)
  * python3-forbiddenfruit
  * r-bio (extends the R container with biology packages)

### CODE REVIEW

File: grading/htmlcss/Dockerfile [hunk 1]
```diff
@@ -5,8 +5,8 @@ ARG   VERSION=latest
 FROM  ingi/inginious-c-base:${VERSION}
 LABEL org.inginious.grading.name="htmlcss"
 
-RUN     yum -y install ant bsdtar java-1.8.0 && \
-        mkdir /opt/vnu && curl -L https://github.com/validator/validator/releases/download/17.9.0/vnu.jar_17.9.0.zip | bsdtar -xf - -C /opt/vnu/ --strip-components 1 && \
+RUN     yum -y install ant bsdtar java-1.8.0-openjdk && \
```
Comment (GuillaumeDerval) [inline grading/htmlcss/Dockerfile:8]: There is a forgotten `yum` here #pedantic

File: grading/oz/Dockerfile [hunk 2]
```diff
@@ -5,24 +5,24 @@ ARG   VERSION=latest
...
+            sox lame-libs
 
-ADD     mozart2-2.0.1+build.8.db41d83-x86_64-centos.rpm /mozart.rpm
-
-RUN     yum -y --nogpgcheck localinstall /mozart.rpm && \
-        rm /mozart.rpm && \
-        yum clean all && \
+RUN     dnf -y --nogpgcheck install https://github.com/mozart/mozart2/releases/download/v2.0.1/mozart2-2.0.1-x86_64-linux.rpm && \
+        dnf clean all && \
         rm -rf /var/cache/yum
```
Comment (GuillaumeDerval) [inline grading/oz/Dockerfile:23]: Is it still named /var/cache/yum with dnf?

Comment (GuillaumeDerval) [reply grading/oz/Dockerfile:23]: (and is this really needed after the clean all?)

Comment (anthonygego) [reply grading/oz/Dockerfile]: I reworked the whole dockerfile as env and aliases were not available as is in ssh mode

**Comment by Drumor** [comment]
tried to build and looks ok.

**Pull request merged**